### DISCUSSION TITLE
different types of comments

### DISCUSSION THREAD

**Discussion by colevandersWands** [RecordReplay/devtools#3202] [Ideas]
It could be useful to have different types of comments, then a user could filter the comments in a replay to find what they need.  some first types that come to mind (with placeholder names):

- _explanation_, written by someone who knows the codebase to explain rather than discuss
- _private_, just visible to the user that wrote them
- _discussion_, like there is already. anyone can discuss something about the replay

this could be useful in a teaching/onboarding scenario where the goal is to explain a program's design instead of debug it:

1. a teacher or dev records the interactions they want to document
2. they add _explanation_ comments describing how the code works and why it was implemented a certain way
3. a student or new project member can step through, studying the code and explanation comments
4. the student/new dev can write _private_ comments in the replay for their own study
5. they can discuss the replay with others using _discussion_ comments

as the codebase changes the replays would go out of date, but until then this could be helpful

**Comment by joebobmiles** [comment]
Why not call _canonical_ comments just 'explanations' or 'explainers'? Seems a bit more intuitive, directly invoking their purpose instead of muddying things.

**Comment by jonbell-lot23** [comment]
Good feedback, thank you!

**Comment by jonbell-lot23** [comment]
March 15, 2022 update:

There's definitely something here. Our current comment system lists everything in one way, in one order. We've talked about adding flexibility and power here. We're not ready to design it yet, but it's definitely something to keep an eye on as we scale.
